### PR TITLE
New Dapr Scenario: Deploy an application using the Dapr cluster extension

### PR DESCRIPTION
**Pull request by zaidmohd**
New Dapr State store scenario 
- Dapr Scenario Readme
- Dapr Scenario automation script
- Python App deployment Yaml
- Order sample json

Add new Developer User Case Index 12
- Update Unified Operations to 13
- Update IoT Use case to 14

### CODE REVIEW

File: azure_arc_k8s_jumpstart/multi_distributions/dapr/dapr_k8s_statestore.sh [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+export appId='<Your Azure service principal name>'
+export password='<Your Azure service principal password>'
+export tenantId='<Your Azure tenant ID>'
+export resourceGroup='<Azure resource group name>'
+export arcClusterName='<The name of your k8s cluster as it will be shown in Azure Arc>'
+export location='<Azure Region>'
+export daprExtensionName='daprextension'
+export redisSKU='Basic'
+export redisVMSize='c0'
+export redisName=secret-store-$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 10 | head -n 1)
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/multi_distributions/dapr/dapr_k8s_statestore.sh:13]: I would add a "do not change comment" like we do in the CAPI script

File: azure_arc_k8s_jumpstart/multi_distributions/dapr/dapr_k8s_statestore.sh [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+az extension update --name "k8s-extension"
+rm extension_output
+fi
+echo ""
+
+echo "Deploying Azure Cache for Redis"
+az group create --location $location --name $resourceGroup
+az redis create --location $location --name $redisName --resource-group $resourceGroup --sku $redisSKU --vm-size $redisVMSize
+
+echo "Installing Dapr extension"
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/multi_distributions/dapr/dapr_k8s_statestore.sh:33]:
```suggestion
echo "Installing Dapr Azure Arc-enabled Kuberntets Cluster extension"
```

File: azure_arc_k8s_jumpstart/multi_distributions/dapr/dapr_k8s_statestore.sh [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+az k8s-extension create --cluster-type connectedClusters --cluster-name $arcClusterName --resource-group $resourceGroup --name $daprExtensionName --extension-type Microsoft.Dapr
+
+echo "Creating the Kubernetes secret"
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/multi_distributions/dapr/dapr_k8s_statestore.sh:36]:
```suggestion
echo "Creating the Kubernetes secret for Redis"
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/day2/_index.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 type: docs
 title: "Unified Operations Use Cases"
 linkTitle: "Unified Operations Use Cases"
-weight: 12
+weight: 13
 description: >-
   Once you have Kubernetes clusters projected into Azure with Azure Arc, you can start to use native Azure tooling to manage the clusters as native Azure resources. The following scenarios show examples of using Azure management tools such as Azure Monitor, GitOps configurations, and Azure Policy.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/day2/_index.md:7]:
```suggestion
  Once you have Kubernetes clusters projected into Azure with Azure Arc, you can start to use native Azure tooling to manage the clusters as native Azure resources. The following scenarios show examples of using Cluster extensions and Azure management tools such as Azure Monitor, GitOps configurations, and Azure Policy.
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/_index.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+---
+type: docs
+title: "Developer Use Cases"
+linkTitle: "Developer Use Cases"
+weight: 12
+description: >-
+  Once you have Kubernetes clusters projected into Azure with Azure Arc, you can start to use native Azure tooling to manage the clusters as native Azure resources. The following scenarios show examples of using developer-centric tools such as Dapr.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/_index.md:7]:
```suggestion
  Once you onboarded your Azure Arc-enabled Kubernetes clusters, you can start to use native Azure tooling to manage the clusters as native Azure resources. The following scenarios show examples of using developer-centric use-cases.
```

Comment (likamrat) [reply docs/azure_arc_jumpstart/azure_arc_k8s/developer/_index.md:7]: I prefer not to be explicit about Dapr

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/_index.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+---
+type: docs
+title: "Dapr"
+linkTitle: "Dapr"
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/_index.md:4]: Might be good to be more explicit here and mention the name of the scenario in the context of Dapr. i.e Dapr state store integration with Redis" or something like tthat

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+
+- Clone the Azure Arc Jumpstart repository
+
+    ```shell
+    git clone https://github.com/microsoft/azure_arc.git
+    ```
+
+- As mentioned, this scenario starts at the point where you already have a connected Kubernetes cluster to Azure Arc.
+
+    ![Existing Azure Arc-enabled Kubernetes cluster](./01.png)
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:27]: Don't update for this scenario but for future scenarios, let's keep the standard of  cluster resource name to "Azure-Arc-Demo"

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 8]
```diff
@@ -0,0 +1,205 @@
+---
+type: docs
+title: "Deploy an application using the Dapr cluster extension for Azure Arc-enabled Kubernetes cluster"
+linkTitle: "Deploy an application using the Dapr cluster extension for Azure Arc-enabled Kubernetes cluster"
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:4]:
```suggestion
title: "Deploy an application using the Dapr Cluster extension for Azure Arc-enabled Kubernetes cluster"
linkTitle: "Deploy an application using the Dapr Cluster extension for Azure Arc-enabled Kubernetes cluster"
```

Comment (likamrat) [reply docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:4]: It's a feature, that's why the capital C

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+weight: 1
+description: >
+---
+
+## Deploy an application using the Dapr cluster extension for Azure Arc-enabled Kubernetes cluster
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:9]:
```suggestion
## Deploy an application using the Dapr Cluster extension for Azure Arc-enabled Kubernetes cluster
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+
+The following Jumpstart scenario will guide you on how to set up [Dapr](https://dapr.io/) for the Azure Arc-enabled Kubernetes cluster. In this scenario, you will enable the Dapr by deploying the Dapr extension on your Azure Arc-enabled Kubernetes cluster. You will also be deploying a Python application that generates messages and a Node application that consumes and persists them on Azure Cache for Redis.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:11]:
```suggestion
The following Jumpstart scenario will guide you on how to set up [Dapr](https://dapr.io/) for the Azure Arc-enabled Kubernetes cluster. In this scenario, you will enable Dapr by deploying the Dapr Cluster extension on your Azure Arc-enabled Kubernetes cluster alongside a Python application that generates messages and a Node.js application that consumes and persists them on [Azure Cache for Redis](https://docs.microsoft.com/azure/azure-cache-for-redis/cache-overview).
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+
+## Automation Flow
+
+For you to get familiar with the automation and deployment flow, below is an explanation.
+
+- User has deployed the Kubernetes cluster and connected it as an Azure Arc-enabled Kubernetes cluster.
+
+- User is editing the environment variables in the below Shell script file (1-time edit) which then be used throughout the configuration.
+
+- User is running the shell script. The script will use the extension management feature of Azure Arc to deploy the Dapr extension.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:78]:
```suggestion
- User is running the shell script. The script will use the extension management feature of Azure Arc to deploy the Dapr Clsuter extension.
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+
+- The script will also create and configure the state store (Azure Cache for Redis), and deploy the Node.js app with the sidecar.
+
+- User is verifying the cluster and ensuring the Dapr extension is deployed.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:82]:
```suggestion
- User is verifying the cluster and ensuring the Dapr Cluster extension is deployed.
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+
+- User is verifying the service and confirming the order was persisted in the state store.
+
+- Additionally, the User will deploy the Python app with the Dapr sidecar and observe messages to confirm successful persistence.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:86]:
```suggestion
- Additionally, the user will deploy the Python app with the Dapr sidecar and observe messages to confirm successful persistence.
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 14]
```diff
@@ -0,0 +1,205 @@
...
+
+## Deployment
+
+To create a new extension Instance, we will use the _k8s-extension create_ command while passing in values for the mandatory parameters. This scenario provides you with the automation to deploy the Dapr extension on your Azure Arc-enabled Kubernetes cluster.
+
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:91]:
```suggestion
To create a new Cluster extension instance, we will use the _k8s-extension create_ command while passing in values for the mandatory parameters. This scenario provides you with the automation to deploy the Dapr Cluster extension on your Azure Arc-enabled Kubernetes cluster.

```

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 15]
```diff
@@ -0,0 +1,205 @@
...
+> **NOTE: Before installing the Dapr extension, make sure that the _kubectl_ context is pointing to your Azure Arc-enabled Kubernetes cluster. To do that, you can refer to the [official Kubernetes documentation](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/) to find the options to change the kubecontext to different Kubernetes clusters.**
+
+![Screenshot showing current kubectl context pointing to K8s cluster](./03.png)
+
+- In the screenshot below, notice how currently there are no extensions installed yet in your Arc-enabled Kubernetes cluster.
+
+    ![Screenshot showing Azure Portal with Azure Arc-enabled Kubernetes resource extensions](./04.png)
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:98]: Check the previous comment on adding comments :-) and update screenshots

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 16]
```diff
@@ -0,0 +1,205 @@
...
+
+- Edit the environment variables [in the script](https://github.com/microsoft/azure_arc/blob/main/azure_arc_k8s_jumpstart/multi_distributions/dapr/dapr_k8s_statestore.sh) to match your environment parameters followed by running the _`. ./dapr_k8s_statestore.sh`_ command.
+
+    ![Screenshot parameter](./05.png)
+
+    ![Screenshot parameter examples](./06.png)
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:104]: Check the previous comment on adding comments :-) and update screenshots

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 17]
```diff
@@ -0,0 +1,205 @@
...
+
+    > **NOTE: The extra dot is due to the shell script having an _export_ function and needs to have the vars exported in the same shell session as the rest of the commands.**
+
+   The script will:
+
+  - Login to your Azure subscription using the SPN credentials
+  - Add or Update your local _k8s-extension_ Azure CLI extension
+  - Create Resource Group, Redis store and Kubernetes Secret for the Redis Key
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:112]:
```suggestion
  - Create Azure resource group, Redis store, and Kubernetes Secret for the Redis Key
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 18]
```diff
@@ -0,0 +1,205 @@
...
+  - Create Dapr extension instance
+  - Configure the state store on Kubernetes and deploy the Node app
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:114]:
```suggestion
  - Create Dapr extension Cluster instance
  - Configure the state store on Kubernetes and deploy the Node.js app
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 19]
```diff
@@ -0,0 +1,205 @@
...
+
+- Once the script will complete it’s run, you can see that Dapr extension is now enabled in the extension tab section of the Azure Arc-enabled Kubernetes cluster resource in Azure. The script usually take 10-20 minutes to complete.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:116]:
```suggestion
- Once the script will complete it’s run, you can see that the Dapr Cluster extension is now enabled in the extension tab section of the Azure Arc-enabled Kubernetes cluster resource in Azure. The script usually takes 10-20 minutes to complete.
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 20]
```diff
@@ -0,0 +1,205 @@
...
+
+    ![Screenshot script completion](./07.png)
+
+    ![Screenshot extension deployment extensions tab](./08.png)
+
+## Dapr Components & Node App
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:122]:
```suggestion
## Dapr Components & Node.js application
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 21]
```diff
@@ -0,0 +1,205 @@
...
+
+- Before validating the App, let’s verify and “zoom in” to the Kubernetes resources deployed by running a few kubectl commands.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:124]:
```suggestion
- Before validating the app, let’s verify and “zoom in” to the Kubernetes resources deployed by running a few _kubectl_ commands.
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 22]
```diff
@@ -0,0 +1,205 @@
...
+    ```shell
+    kubectl get pods -n dapr-system
+    kubectl get component
+    ```
+
+    ![Screenshot Dapr pods ](./09.png)
+
+    ![Screenshot state store ](./10.png)
+
+- Show the Node App and Service, and make note of the External IP of the service.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:135]:
```suggestion
- Show the Node.js app and Service, and make note of the External IP of the service.
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 23]
```diff
@@ -0,0 +1,205 @@
...
+
+    ```shell
+    kubectl get pods,svc
+    ```
+
+    ![Screenshot Node App pod and service](./11.png)
+
+    > **NOTE: The Dapr control plane will automatically inject the Dapr sidecar to the Pod, as the "dapr.io/enabled: true" annotation is added to the Node App deployment.**
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:143]:
```suggestion
    > **NOTE: The Dapr control plane will automatically inject the Dapr sidecar to the Pod, as the "dapr.io/enabled: true" annotation is added to the Node.js app deployment.**
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 24]
```diff
@@ -0,0 +1,205 @@
...
+
+- Validate the order was persisted by requesting from the app.
+
+    ```shell
+    curl $EXTERNAL_IP/order
+    ```
+
+    ![Screenshot validate order](./14.png)
+
+## Deploy the Python app with the Dapr sidecar
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:170]:
```suggestion
## Deploy the Python application with the Dapr sidecar
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 25]
```diff
@@ -0,0 +1,205 @@
...
+
+This is a basic [Python app](https://github.com/dapr/quickstarts/tree/master/tutorials/hello-world) that posts JSON messages to localhost:3500, which is the default listening port for Dapr. You can invoke the Node.js application's neworder endpoint by posting to v1.0/invoke/nodeapp/method/neworder. The message contains some data with an orderId that increments once per second.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:172]:
```suggestion
The following is a basic "hello world" [Python app](https://github.com/dapr/quickstarts/tree/master/tutorials/hello-world) that posts JSON messages to localhost:3500, which is the default listening port for Dapr. You can invoke the Node.js application's _neworder_ endpoint by posting to _v1.0/invoke/nodeapp/method/neworder_. The message contains some data with an _orderId_ that increments once per second.
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md [hunk 26]
```diff
@@ -0,0 +1,205 @@
...
+    kubectl logs --selector=app=python -c daprd --tail=-1
+    ```
+
+    ![Screenshot validate messages and API calls](./16.png)
+
+    ![Screenshot validate messages and API calls](./17.png)
+
+    ![Screenshot validate messages and API calls](./18.png)
+
+- Call the Node app's order endpoint to get the latest order and confirm successful persistence. You should see the latest JSON in the response.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:199]:
```suggestion
- Call the Node.js app's order endpoint to get the latest order and confirm successful persistence. You should see the latest JSON in the response.
```

**Review by likamrat — CHANGES_REQUESTED** [review]
Clean work, very nice! Few comments:

- Add a cleanup section in the end like with other scenarios
- Add couple of comments in the script and update screenshot accordenly 
- Some suggestions around lingo

File: azure_arc_k8s_jumpstart/multi_distributions/dapr/dapr_k8s_statestore.sh (hunk 1 above)
Comment (zaidmohd) [reply azure_arc_k8s_jumpstart/multi_distributions/dapr/dapr_k8s_statestore.sh:13]: Added

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md (hunk 7 above)
Comment (zaidmohd) [reply docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:27]: Updated

File: docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md (hunk 16 above)
Comment (zaidmohd) [reply docs/azure_arc_jumpstart/azure_arc_k8s/developer/dapr/dapr_statestore/_index.md:104]: Updated

**Review by likamrat — APPROVED** [review]
LGTM

**Pull request merged**